ui/src/app/applications/components/application-details/application-resource-list.tsx

```diff
@@ -6,88 +6,121 @@ import {ResourceIcon} from '../resource-icon';
 import {ResourceLabel} from '../resource-label';
 import {ComparisonStatusIcon, HealthStatusIcon, nodeKey} from '../utils';
 import {Consumer} from '../../../shared/context';
+import {services} from '../../../shared/services';
 
 export const ApplicationResourceList = ({
     resources,
     onNodeClick,
-    nodeMenu
+    nodeMenu,
+    application
 }: {
-    resources: models.ResourceStatus[];
+    resources: any[];
```

Comment:
Since this will be done after each render, this column will populate after some delay. It flashes at times.
I'm wondering if the sync wave information can be determined beforehand.  Have you looked at DataLoader?

Reply:
Great point! Thanks for the advice. I moved the service call in DataLoader in `application-details` component.

Reply:
Hi @ciiay , could the data loading apply only to the list view?  There will be extra loading of the sync information for other views, for example.

Reply:
Thanks @keithchong for the catch, code updated to improve this behavior.